src/SharpNeat/NeuralNets/Double/ActivationFunctions/BitwiseReLU.cs

```diff
@@ -0,0 +1,73 @@
+﻿// This file is part of SharpNEAT; Copyright Colin D. Green.
+// See LICENSE.txt for details.
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+namespace SharpNeat.NeuralNets.Double.ActivationFunctions;
+
+/// <summary>
+/// Rectified linear activation unit (ReLU) using bitwise operators.
+/// </summary>
+public sealed class BitwiseReLU : IActivationFunction<double>
+{
+    /// <inheritdoc/>
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    public void Fn(ref double x)
+    {
+        long sign = BitConverter.DoubleToInt64Bits(x) >> 63;
+        x *= (sign & (-sign >> 63)) | (~sign & 1);
```

Comment:
I think `(sign & (-sign >> 63))` always evaluate to zero and is therefore redundant(?)

Reply:
You're right, I've removed that section and its still passing my tests.

Reply:
Here's an even faster version that avoids the multiplication:

```
    private static void BitwiseFn2(ref double x)
    {
        long xlong = BitConverter.DoubleToInt64Bits(x);
        x = BitConverter.Int64BitsToDouble(xlong & ~(xlong >> 63));
    }
```

For JITter output see:
https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIAYACY8gOgDkBXfGKASzFwBuGjWIBmJqQYBhBgG8aDJQ0XKADnwBu2DDCbkkTFAwBCvDAHdeuGADEAdgApYAMwYATCB2AAbPQgBKVSUFamVwhh8IewBzBlxeGPsGAF5Tc2lozR5dKBYAES9fGAAVCABJewxUMwxcR0CGAD4mhiQxYTCIpQQGACo0xwA/BKSGADIGcgDO8IBfcOCVLqUNXm1dfUNiY1qrGwdSZxg3T28/BkCl0O6lKNjL+7i02sz7bKhcgqK/MsrqlC1eqBWa3XovDJZHI8Fj/GrmXBlQrnGANJ4TBhDNHROItNpiAIzJZzEQrBhrDZ6ZjbYwOY6nH7+IJkm63XguBoMAA8DDoLDozNuEXBvP5oKUJOocyAA===